ContractsApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs

```diff
@@ -206,17 +206,47 @@ public async Task PatchContractReturnsNullIfContractNotFound()
 
             response.Should().BeNull();
         }
+        [Fact]
```

Comment:
Do we have a test for if one or both dates are null?

Reply:
Thanks for the valid point. After much thinking, this is the change I've implemented in the gateway:
`            if (contractRequestBody.HandbackDate != null) //this is because a patch might not contain this date
            {
                if (existingContract.StartDate > contractRequestBody.HandbackDate || existingContract.StartDate is null)
                    throw new StartAndHandbackDatesConflictException(existingContract.StartDate, contractRequestBody.HandbackDate);
            }`
So, if both dates come in via the request, they are checked by the FV validator. If only the Handback date comes in, and:
- is prior to the saved Start date 
- or if the saved Start date is null
the validation will throw the exception.
Hope it makes sense!